source/common/http/http1/conn_pool.cc

```diff
@@ -273,17 +276,19 @@ ConnPoolImpl::StreamWrapper::~StreamWrapper() {
 void ConnPoolImpl::StreamWrapper::onEncodeComplete() { encode_complete_ = true; }
 
 void ConnPoolImpl::StreamWrapper::decodeHeaders(HeaderMapPtr&& headers, bool end_stream) {
-  if (headers->Connection() &&
-      absl::EqualsIgnoreCase(headers->Connection()->value().getStringView(),
-                             Headers::get().ConnectionValues.Close)) {
-    saw_close_header_ = true;
-    parent_.parent_.host_->cluster().stats().upstream_cx_close_notify_.inc();
+  if (headers->Connection()) {
+    if (absl::EqualsIgnoreCase(headers->Connection()->value().getStringView(),
```

Comment:
perf nit: can you check protocol in each of these if cases so we short circuit the connection check depending on the protocol?

Reply:
The logic has changed with the protocol checks.  PTAL